jaworski_cellseg/label_counting_widget.py

```diff
@@ -1,9 +1,53 @@
 from magicgui import magicgui
 import napari
 from skimage.measure import regionprops
+import pandas as pd
 
 
-def create_label_counting_widget(viewer: napari.Viewer):
+def generate_statistics(regions, physical_sizes, save_file_path):
+    # Initialize a list to hold the properties data
+    data = []
+
+    # Loop through each region and filter for the labels of interest
+    for region in regions:
```

Comment:
Initializing empty lists and populating them in a for loop with `.append is slow`. If you refactor the business logic in to functions consider using a comprehension.

Reply:
I modified the code to improve readability and performance